<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>ControlPanel</class>
    <widget class="QWidget" name="ControlPanel">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>300</width>
                <height>750</height>
            </rect>
        </property>
        <property name="windowTitle">
            <string>Control Panel</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
            <item>
                <widget class="QGroupBox" name="setupBox">
                    <property name="title">
                        <string>Graph settings</string>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayout_2">
                        <item>
                            <widget class="QPushButton" name="setupGraphButton">
                                <property name="text">
                                    <string>Create new graph</string>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <item>
                <widget class="QGroupBox" name="graphEditBox">
                    <property name="title">
                        <string>Edit graph</string>
                    </property>
                    <layout class="QGridLayout" name="gridLayout">
                        <item row="0" column="0">
                            <widget class="QPushButton" name="addVertexButton">
                                <property name="text">
                                    <string>Add Vertex</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="1">
                            <widget class="QPushButton" name="addEdgeButton">
                                <property name="text">
                                    <string>Add Edge</string>
                                </property>
                            </widget>
                        </item>
                        <item row="1" column="0">
                            <widget class="QPushButton" name="removeVertexButton">
                                <property name="text">
                                    <string>Remove Vertex</string>
                                </property>
                            </widget>
                        </item>
                        <item row="1" column="1">
                            <widget class="QPushButton" name="removeEdgeButton">
                                <property name="text">
                                    <string>Remove Edge</string>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <item>
                <widget class="QGroupBox" name="algorithmBox">
                    <property name="title">
                        <string>Algorithm</string>
                    </property>
                    <layout class="QFormLayout" name="formLayout">
                        <item row="0" column="0">
                            <widget class="QLabel" name="labelAlgorithm">
                                <property name="text">
                                    <string>Algorithm:</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="1">
                            <widget class="QComboBox" name="algorithmComboBox"/>
                        </item>
                        <item row="1" column="0">
                            <widget class="QLabel" name="labelStartVertex">
                                <property name="text">
                                    <string>Start vertex:</string>
                                </property>
                            </widget>
                        </item>
                        <item row="1" column="1">
                            <widget class="QComboBox" name="startVertexComboBox"/>
                        </item>
                    </layout>
                </widget>
            </item>
            <item>
                <widget class="QGroupBox" name="playerBox">
                    <property name="title">
                        <string>Player Controls</string>
                    </property>
                    <layout class="QGridLayout" name="playerLayout">

                        <item row="0" column="0">
                            <widget class="QPushButton" name="prevButton">
                                <property name="text">
                                    <string>&lt;&lt; Prev</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="1">
                            <widget class="QPushButton" name="startButton">
                                <property name="text">
                                    <string>Start</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="2">
                            <widget class="QPushButton" name="playButton">
                                <property name="text">
                                    <string>▶ Play</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="3">
                            <widget class="QPushButton" name="pauseButton">
                                <property name="text">
                                    <string>⏸ Pause</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="4">
                            <widget class="QPushButton" name="nextButton">
                                <property name="text">
                                    <string>Next &gt;&gt;</string>
                                </property>
                            </widget>
                        </item>
                        <item row="1" column="0" colspan="5">
                            <widget class="QSlider" name="speedSlider">
                                <property name="orientation">
                                    <enum>Qt::Horizontal</enum>
                                </property>
                                <property name="minimum">
                                    <number>1</number>
                                </property>
                                <property name="maximum">
                                    <number>100</number>
                                </property>
                                <property name="value">
                                    <number>50</number>
                                </property>
                                <property name="toolTip">
                                    <string>Adjust playback speed</string>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <item>
                <widget class="QPushButton" name="resetButton">
                    <property name="text">
                        <string>Reset</string>
                    </property>
                </widget>
            </item>
            <item>
                <spacer name="verticalSpacer">
                    <property name="orientation">
                        <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                        <size>
                            <width>20</width>
                            <height>40</height>
                        </size>
                    </property>
                </spacer>
            </item>
        </layout>
    </widget>
    <resources/>
    <connections/>
</ui>